DependencyInjection/Configuration.php

```
@@ -21,20 +21,49 @@ public function getConfigTreeBuilder()
         $treeBuilder = new TreeBuilder();
         $treeBuilder
             ->root('nelmio_api_doc')
+            ->beforeNormalization()
+                ->ifTrue(function ($v) {
+                    return !isset($v['areas']) && isset($v['routes']);
```

if both areas and routes are set, we should throw an invalid configuration exception explaining to use only the new `areas` section. otherwise the user will just see an error that routes is not known, which will be confusing when before, routes did work.